File: androidApp/src/main/java/com/example/ula_app/android/ui/welcome/WelcomePage2.kt
package com.example.ula_app.android.ui.welcome

import android.util.Log
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material.Icon
import androidx.compose.material.MaterialTheme
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.scale
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.dp
import com.example.ula_app.android.R
import com.example.ula_app.android.ui.component.IconButton

@Composable
fun WelcomePage2(
    onPreviousButtonClicked: () -> Unit,
    onNextButtonClicked: () -> Unit
) {
    // Read icon and images from drawable folder.
    val nextButtonIcon = painterResource(id = R.drawable.next)
    val previousButtonIcon = painterResource(id = R.drawable.back)
    val arrowIcon = painterResource(id = R.drawable.arrows)
    val adultNormalMonster = painterResource(id = R.drawable.ic_stickman_4)
    val adultFitMonster = painterResource(id = R.drawable.stickman_draft_monster_5aug2014)
    val adultSleepingMonster = painterResource(id = R.drawable.bed)
    val adultFatMonster = painterResource(id = R.drawable.stickman_fat)


    val leftPartWeight = 4.5f
    val middlePartWeight = 1f
    val rightPartWeight = 4.5f

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Color.White),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.SpaceBetween
    ) {
        Column(
            modifier = Modifier.padding(
                start = Default.padding.start,
                end = Default.padding.end,
                top = Default.padding.top,
                bottom = Default.padding.bottom,
            )
        ) {
            Row(
                modifier = Modifier
                    .fillMaxWidth(),
                horizontalArrangement = Arrangement.Center
            ) {
                Text(
                    text = "Ok?",
                    style = MaterialTheme.typography.h1
                )
            }

            Spacer(modifier = Modifier.height(60.dp))

            Row (
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.Center
            ) {
                Column(
                    modifier = Modifier.weight(leftPartWeight),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Icon(
                        painter = adultNormalMonster,
                        contentDescription = null,
                        modifier = Modifier.scale(1.4f)
                    )
                }

                Column(
                    modifier = Modifier.weight(middlePartWeight),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Icon(
                        painter = arrowIcon,
                        contentDescription = null
                    )
                }

                Column(
                    modifier = Modifier.weight(rightPartWeight),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Icon(
                        painter = adultFitMonster,
                        contentDescription = null
                    )
                }

            }

            Spacer(modifier = Modifier.height(60.dp))

            Row (
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.Center
            ) {
                Column(
                    modifier = Modifier.weight(leftPartWeight),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Icon(
                        painter = adultSleepingMonster,
                        contentDescription = null,
                        modifier = Modifier.scale(0.75f)
                    )
                }

                Column(
                    modifier = Modifier.weight(middlePartWeight),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Icon(
                        painter = arrowIcon,
                        contentDescription = null
                    )
                }

                Column(
                    modifier = Modifier.weight(rightPartWeight),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Icon(
                        painter = adultFatMonster,
                        contentDescription = null,
                        modifier = Modifier.scale(0.5f)
                    )
                }

            }
        }

        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            IconButton(
                Icon = {
                    Icon(
                        painter = previousButtonIcon,
                        contentDescription = "Previous"
                    )
                },
                onClick = {
                    onPreviousButtonClicked()
                    Log.i("WelcomePage_Page2", "Click Back on the Page 2")
                }
            )
            IconButton(
                Icon = {
                    Icon(
                        painter = nextButtonIcon,
                        contentDescription = "Next"
                    )
                },
                onClick = {
                    onNextButtonClicked()
                    Log.i("WelcomePage_Page2", "Click Next on the Page 2")
                }
            )
        }
    }
}

File: iosApp/iosApp/welcome/ui/WelcomePage2.swift
//
//  WelcomePage2.swift
//  iosApp
//
//  Created by Hao Wu on 9/18/23.
//  Copyright © 2023 orgName. All rights reserved.
//

import Foundation
import SwiftUI

var leftPartWeight = 4.5
var middlePartWeight = 1.0
var rightPartWeight = 4.5

struct WelcomePage2: View {
    var onPreviousButtonClicked: () -> Void = {}
    var onNextButtonClicked: () -> Void = {}
    
    var body: some View {
        VStack(alignment: .center) {
            VStack {
                HStack {
                    Text("Ok?")
                        .font(.WelcomeTitle)
                        .foregroundColor(.red)
                }
                
                Spacer().frame(height: 60)
                
                HStack {
                    VStack {
                        Image("ic_stickman_4")
                            .resizable()
                            .frame(width: 120, height: 90)
                    }
                    VStack {
                        Image("arrows")
                            .resizable()
                            .frame(width: 50, height: 25)
                    }
                    VStack {
                        Image("stickman_draft_monster_5aug2014")
                            .resizable()
                            .frame(width: 60, height: 80)
                    }
                }
                
                Spacer().frame(height: 60)
                
                HStack {
                    VStack {
                        Image("bed")
                            .resizable()
                            .frame(width: 120, height: 50)
                    }
                    VStack {
                        Image("arrows")
                            .resizable()
                            .frame(width: 50, height: 25)
                    }
                    VStack {
                        Image("stickman_fat")
                            .resizable()
                            .frame(width: 50, height: 70)
                    }
                }
            }
            .padding(.leading, 20)
            .padding(.trailing, 20)
            .padding(.top, 50)
            .padding(.bottom, 0)
            
            Spacer()
            
            HStack() {
                BackButton(
                    onClick: {
                        onPreviousButtonClicked()
                    }
                )
                
                Spacer()
                
                NextButton(
                    onClick: {
                        onNextButtonClicked()
                    }
                )
            }
            .padding(.horizontal, 10)
        }
    }
}

struct WelcomePage2_Previews: PreviewProvider {
    static var previews: some View {
        WelcomePage2(
            onPreviousButtonClicked: {},
            onNextButtonClicked: {}
        )
    }
}
